### PR TITLE
[Feature Request]: Add Leetcode solution 15-17 #553

### PR DESCRIPTION
**Pull request by nishant0708**
Fixed: #553 

## Description

Added the solution of leetcode problems from 15-17

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x ] I have read and followed the Contribution Guidelines.
- [ ] I have tested the changes thoroughly before submitting this pull request.
- [x ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/0000-0099/0017-Letter Combinations of a Phone Number.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+---
+id: Letter Combinations of a Phone Number
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0017-Letter Combinations of a Phone Number.md:2]: replace `id: Letter Combinations of a Phone Number` to `id: letter-combinations-of-a-phone-number`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: dsa-solutions/lc-solutions/0000-0099/0017-Letter Combinations of a Phone Number.md (hunk 1 above)
Comment (nishant0708) [reply dsa-solutions/lc-solutions/0000-0099/0017-Letter Combinations of a Phone Number.md:2]: Okay I will do this

**Comment by nishant0708** [comment]
hey @Ajay-Dhangar  please check it now

**Comment by Ajay-Dhangar** [comment]
> hey @Ajay-Dhangar please check it now

push latest code for this file in pr

**Comment by nishant0708** [comment]
sorry @Ajay-Dhangar  i pushed code on different branch .and asked you for review now it will be visible to you

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Review by Ajay-Dhangar — APPROVED** [review]
Done and Fixed issue

**Pull request merged**